app/services/post_status_service.rb

```diff
@@ -23,13 +23,16 @@ def call(account, text, in_reply_to = nil, **options)
     status = nil
     text   = options.delete(:spoiler_text) if text.blank? && options[:spoiler_text].present?
 
+    visibility = options[:visibility] || account.user&.setting_default_privacy
+    visibility = :unlisted if visibility == :public && account.silenced
+
     ApplicationRecord.transaction do
       status = account.statuses.create!(text: text,
                                         media_attachments: media || [],
                                         thread: in_reply_to,
                                         sensitive: (options[:sensitive].nil? ? account.user&.setting_default_sensitive : options[:sensitive]) || options[:spoiler_text].present?,
                                         spoiler_text: options[:spoiler_text] || '',
-                                        visibility: options[:visibility] || account.user&.setting_default_privacy,
+                                        visibility: visibility,
```

Comment:
This does give away to the user that they are silenced. I thought you want to hook into the ActivityPub-facing JSON to force it there. Although you could argue that they can look at the ActivityPub JSON likewise to find out, so maybe this is the simplest solution...

Reply:
They can already to this by looking at the public timeline of their instance…?
Also, I'm not sure I see the point of silently silencing a local user. Shouldn't a decision about a *local* user be known to them?